### PR TITLE
CLOUDP-51068: add mongosqld.gz as valid option for mcli atlas logs download

### PR DESCRIPTION
**Pull request by shalka**
## Proposed changes

_Jira ticket:_ CLOUDP-51068

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by gssbzn** [comment]
Thanks for promptly adding this to the cli, one thing is we are missing an e2e test for this and we currently test all log types [here](https://github.com/mongodb/mongocli/blob/master/e2e/atlas/logs_test.go) 
I'm not sure if the API will work ok if there's no BI connector configured so up to you if we may need to run that test with a cluster running BI but I'd like to have an e2e test for this case as we are already testing the rest

Also please note that we don't run tests that require clusters running by default on PRs so please either attach an evergreen patch with the logs tests passing or you can modify the PR patch to include the logs test

Configure-Patch

Finally I'm still OoO this week so I'll delegate this

**Review by andreaangiolillo — CHANGES_REQUESTED** [review]
Hello! Thanks for taking the time to make this change.
Unfortunately, this mongocli command calls this [endpoint](https://docs.atlas.mongodb.com/reference/api/logs) which doesn’t support` mongosqld.gz` for now.

**Comment by gssbzn** [comment]
@shalka in #603 I have added an flag to deploy the biconnector when creating a cluster and also made that enabled by default on our cluster helper to help unblock this let me know if I can help you further

**Comment by shalka** [comment]
Thanks @gssbzn let me rebase and get some tests on this branch! I should be able to fix up this PR this week

**Comment by shalka** [comment]
@gssbzn thanks for adding the biConnector flag to the default cluster configuration! unfortunately, there's some missing functionality and i look to you to tell me how you'd like me to proceed.

```
> mongocli atlas -P local logs download mongosqld.gz
Error: "mongocli atlas logs download" requires 2 arguments

Usage:  mongocli atlas logs download <hostname> <mongodb.gz|mongos.gz|mongosqld.gz|mongodb-audit-log.gz|mongos-audit-log.gz> [flags]
> mongocli atlas -P local logs download cloudp-51068-mcli-02-biconnector.p9xkj.mmscloudteam.com mongosqld.gz
Download of mongosqld.log.gz completed.
```

I was able to discover the hostname for the biConnector in the UI:
Screen Shot 2020-12-16 at 10 40 20 AM

But I'm not sure how I can programmatically discover the biConnector using the existing CLI commands (making testing this feature impossible, presently). Unless you know of another way to surface hostnames of the underlying cluster? 

When calling `mongocli atlas processes list` I'm also unable to find the required information (which I think we should file a defect for, but I look to your knowledge of CLI first before I file it):

```
> mongocli atlas -P local processes list
ID                                                      REPLICA SET NAME       SHARD NAME   VERSION
atlas-ewvih0-shard-00-00.p9xkj.mmscloudteam.com:27017   atlas-ewvih0-shard-0                4.4.4
atlas-ewvih0-shard-00-01.p9xkj.mmscloudteam.com:27017   atlas-ewvih0-shard-0                4.4.4
atlas-ewvih0-shard-00-02.p9xkj.mmscloudteam.com:27017   atlas-ewvih0-shard-0                4.4.4
```
Separately, I was able to download the biConnector logs using a local cluster with biConnector enabled and confirm they are correct by comparing against the same logs downloaded via the cluster card UI. This CLI functionality is working, though it doesn't seem useful to me if I cannot discover the biConnector hostname. Let me know how we should proceed.

My thinking is that we should ship this functionality, and file a separate ticket to surface the biConnector hostname.

**Comment by gssbzn** [comment]
@shalka sure, let's rollback the e2e changes so they don't fail and follow up on how we can expose the bi connector host via api as right now it seems a bit inconvenient for cli users to find that out

**Comment by shalka** [comment]
@gssbzn i think failing tests aren't related to this change and i haven't made e2e changes part of this PR. please approve and we can sync offline to talk about followup tickets that will enrich CLI functionality, as well as provide testability?

File: e2e/atlas/logs_test.go
Comment (gssbzn) [inline e2e/atlas/logs_test.go:51]: > i haven't made e2e changes part of this PR

@shalka this change is the one I refer to, this will fail once it gets merged, right?

Comment (shalka) [reply e2e/atlas/logs_test.go]: You are exactly right! This was an oversight on my part and I've pushed a correction.

**Review by gssbzn — APPROVED** [review]
LGTM thanks for the addition

**Pull request merged**
